sdk/deploy_test.go

```diff
@@ -0,0 +1,140 @@
+
+		s := testCreateResponseServer(t, projectRequestAssertions, projectsResponseBody, 200)
+		defer s.Close()
+		client := testCreateDeployClient(t, s.URL)
+
+		history, err := client.GetHistory(DeployHistoryQuery{ProjectID: "project-NaN"})
+		require.Nil(t, history)
+		require.EqualError(t, err, fmt.Sprintf("%s: json: cannot unmarshal number into Go struct field Project._id of type string", ErrGeneric))
+		require.True(t, errors.Is(err, ErrGeneric))
+	})
+
+	t.Run("Error occurs when projectId does not exist in download list", func(t *testing.T) {
+		projectsResponseBody := `[{"_id":"mongo-id-1","name":"Project 1","configurationGitPath":"/clients/path","projectId":"project-1","environments":[{"label":"Development","value":"development","cluster":{"hostname":"127.0.0.1","namespace":"project-1-dev"}}],"pipelines":{"type":"gitlab"}},{"_id":"mongo-id-2","name":"Project 2","configurationGitPath":"/clients/path/configuration","projectId":"project-2","environments":[{"label":"Development","value":"development","cluster":{"hostname":"127.0.0.1","namespace":"project-2-dev"}},{"label":"Production","value":"production","cluster":{"hostname":"127.0.0.1","namespace":"project-2"}}]}]`
+
+		s := testCreateResponseServer(t, projectRequestAssertions, projectsResponseBody, 200)
+		defer s.Close()
+		client := testCreateDeployClient(t, s.URL)
+
+		history, err := client.GetHistory(DeployHistoryQuery{ProjectID: "project-NaN"})
+		require.Nil(t, history)
+		require.EqualError(t, err, fmt.Sprintf("%s: project-NaN", ErrProjectNotFound))
+		require.True(t, errors.Is(err, ErrProjectNotFound))
+	})
+
+	t.Run("Error occurs when downloading deploy history", func(t *testing.T) {
+		projectsResponseBody := `[{"_id":"mongo-id-1","name":"Project 1","configurationGitPath":"/clients/path","projectId":"project-1","environments":[{"label":"Development","value":"development","cluster":{"hostname":"127.0.0.1","namespace":"project-1-dev"}}],"pipelines":{"type":"gitlab"}},{"_id":"mongo-id-2","name":"Project 2","configurationGitPath":"/clients/path/configuration","projectId":"project-2","environments":[{"label":"Development","value":"development","cluster":{"hostname":"127.0.0.1","namespace":"project-2-dev"}},{"label":"Production","value":"production","cluster":{"hostname":"127.0.0.1","namespace":"project-2"}}]}]`
+		historyResponseBody := `{"statusCode":500,"error":"InternalServerError","message":"some server error"}`
+		responses := []response{
+			{assertions: projectRequestAssertions, body: projectsResponseBody, status: 200},
+			{assertions: historyRequestAsserions, body: historyResponseBody, status: 500},
+		}
+		s := testCreateMultiResponseServer(t, responses)
+		defer s.Close()
+
+		client := testCreateDeployClient(t, s.URL)
+
+		history, err := client.GetHistory(DeployHistoryQuery{ProjectID: "project-2"})
+		require.Nil(t, history)
+		require.NotNil(t, err)
+	})
+
+	t.Run("Error on malformed history items", func(t *testing.T) {
```

Comment:
Also here, write why it's malformed history items

Reply:
It's in the error assertion but OK

```suggestion
	t.Run("Error on malformed history items (invalid DeployItem.ID)", func(t *testing.T) {
```